js/localgov_directories_entry_search.js

```diff
@@ -8,30 +8,33 @@
   Drupal.behaviors.localgovDirectoriesSearch = {
     attach: function attach(context, settings) {
       // Build a select list with options from all the search boxes.
+      var form_ids = Object.keys(drupalSettings.localgovDirectories.directoriesSearch);
       var channelsDropdown = document.createElement('select');
-      Object.keys(drupalSettings.localgovDirectories.directoriesSearch).forEach( function(form_id) {
-        var channel = document.createElement('option');
-        channel.value = form_id;
-        channel.text = Drupal.checkPlain(drupalSettings.localgovDirectories.directoriesSearch[form_id]);
-        channelsDropdown.appendChild(channel);
-      });
-      // Swap the select list into the title.
-      Object.keys(drupalSettings.localgovDirectories.directoriesSearch).forEach( function(form_id) {
-        var label = document.getElementById(form_id + '--channel');
-        label.innerHTML = channelsDropdown.outerHTML;
-        label.childNodes[0].value = form_id;
-        // With an event that hides the current, unhides the selected,
-        // and keeps the value of the selectors correct for the search
-        // they are on.
-        label.childNodes[0].addEventListener('change', function() {
-          var previousId = label.id.slice(0, -9);
-          var previous = document.getElementById(previousId);
-          previous.style.display = 'none';
-          var selected = document.getElementById(label.childNodes[0].value);
-          selected.style.display = 'block';
-          label.childNodes[0].value = previousId;
+      if (form_ids.length > 1) {
```

Comment:
Err.. I would just do 
```
if (form_ids.length === 1) {
  return;
}
```
and then leave the rest of the code undisturbed to avoid adding more indentation to the main logic.  Not a big deal.

Reply:
The early returns debate :) 

Might be more readable if it's obvious here, but as this has been waiting to merge for ages I'm just going to do it.